lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -151,8 +152,50 @@ def which(cmd):
     return None
 
 
-class MPI(object):
+class RunFileLoopMPI(RunFile, RunLoop):
+    """
+    Combination of RunFile and RunLoop to support output to file,
+    while also checking whether any output has been produced after a specified amount of time.
+    """
+    def __init__(self, cmd, **kwargs):
+        """
+        handle initialisation: get filename and output timeout from arguments
+        """
+        self.filename = kwargs.get('filename', None)
+        self.output_timeout = kwargs.pop('output_timeout', None)
+        super(RunFileLoopMPI, self).__init__(cmd, **kwargs)
+
+        self.timeout_cnt = 1
+        self.seen_output = False
+
+    def _loop_process_output(self, output):
+        """
+        check if process is generating any output at all; if not, warn the user after a set amount of time
+        """
+        if output:
+            print "Output was found using RunFile:\n%s\n This means something went horribly wrong." % output
 
+        if self.seen_output:
+            return
+        else:
```

Comment:
since you `return` in the `if`, you don't need to put this in an `else` (less indent)